mozetl/maudau/maudau.py

```diff
@@ -0,0 +1,214 @@
+"""
+Calculate Firefox monthly- and daily-active users.
+"""
+import csv
+import datetime as DT
+import logging
+import os
+
+from pyspark.sql import SparkSession
+import boto3
+import botocore
+import click
+from boto3.s3.transfer import S3Transfer
+from moztelemetry.standards import (
+    filter_date_range,
+    count_distinct_clientids
+)
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+DEVELOPMENT = False
+
+STORAGE_BUCKET = "net-mozaws-prod-us-west-2-pipeline-analysis"
+STORAGE_SUB_DIR = 'spenrose/maudau' if DEVELOPMENT else 'mreid/maudau'
+MAUDAU_ROLLUP_BASENAME = "engagement_ratio.csv"
+MAUDAU_SNAPSHOT_TEMPLATE = "engagement_ratio.{}.csv"
+DASHBOARD_BUCKET = "net-mozaws-prod-metrics-data"
+DASHBOARD_KEY = "firefox-dashboard/{}".format(MAUDAU_ROLLUP_BASENAME)
+
+today = DT.date.today()
+yesterday = today - DT.timedelta(1)
```

Comment:
This ETL job is a bit unusual in that it does not target a specific day. It looks at the existing data and determines *all* the days that need to be re-computed and updates each one that needs to be updated. This is different from how most of our airflow jobs work, but I think we should land it with the current logic first, then figure out how to refactor it to be more friendly to backfill operations / incremental dataset generation in a subsequent PR.